meteor/server/publications/pieceContentStatusUI/checkPieceContentStatus.ts

```diff
@@ -510,13 +514,14 @@ async function checkPieceContentExpectedPackageStatus(
 		for (const expectedPackage of piece.expectedPackages) {
 			// Route the mappings
 			const routedDeviceIds = routeExpectedPackage(routes, studio.mappings, expectedPackage)
+			const packageContainers = applyAndValidateOverrides(studio.packageContainersWithOverrides).obj
```

Comment:
This is still deeper inside loops than I would like.

I think the best thing to do would be to do the same as was done for `mappings` and `routeSets` in the `PieceContentStatusStudio` type. That ensures it will only be done once when constructing that studio object, and avoids needing to pass around another parameter all over the place.

Reply:
Hopefully it just be good now :)